torch/nativert/executor/PlacementUtils.cpp

```diff
@@ -0,0 +1,37 @@
+#include <torch/nativert/executor/Placement.h>
+
+#include <fmt/ostream.h>
+
+namespace torch::nativert {
+
+c10::Device normalizeDevice(const c10::Device& device) {
+  // cpu device doesn't have index
+  // cuda device index must have a index
+  if (device.is_cpu()) {
+    return c10::Device(c10::DeviceType::CPU);
+  } else if (device.is_cuda()) {
+    return c10::Device(
+        c10::DeviceType::CUDA,
+        device.has_index() ? device.index() : static_cast<c10::DeviceIndex>(0));
+  } else {
+    TORCH_CHECK(false, "Unsupported device type", device);
+  }
+}
+
+bool isSameDevice(const c10::Device& a, const c10::Device& b) {
```

Comment:
Here only `cpu` and `cuda` are mentioned, how about other devices? Should we take this opportunity to enhance it?

Reply:
> Here only `cpu` and `cuda` are mentioned, how about other devices? Should we take this opportunity to enhance it?

@cyyever  I see in the original code D56254093 that other devices are intentionally left behind until we introduce their backend.